app/models/session.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Contains a token to identify users
+class Session
+  include ActiveModel::Validations
+  include ActiveModel::Serialization
+
+  def initialize(session_params)
+    @email = session_params[:email]
+    @password = session_params[:password]
+  end
+
+  validates :email, presence: true
+  validates :password, presence: true
+  validate :authenticated
+
+  def token
+    JwtCodec.new.encode({ id: user.id })
+  end
+
+  private
+
+  attr_reader :email, :password
+
+  def user
+    @user ||= User.find_by(email: @email)&.authenticate(@password)
+  end
+
+  def authenticated
+    errors.add :base, :authenticated if user.blank?
```

Comment:
Why don't you create a separate validator class?

Reply:
IMO it's an overkill in this case.

Reply:
Anyway, `Session` is to be refactored in #9 to be testable. I moved this logic to another class and I will apply your suggestion to the other class. Thanks